CRM/Core/BAO/Address.php

```diff
@@ -537,7 +537,7 @@ public static function &getValues($entityBlock, $microformat = FALSE, $fieldName
       $values['display'] = $address->display;
       $values['display_text'] = $address->display_text;
 
-      if (is_numeric($address->master_id)) {
+      if (isset($params['master_id']) && !CRM_Utils_System::isNull($address->master_id)) {
```

Comment:
Wait now you've changed the meaning of this conditional. Are you sure about that?

Reply:
No... I've fixed that now.